modules/siddhi-core/src/main/java/org/wso2/siddhi/core/aggregation/IncrementalDataPurging.java

```diff
@@ -0,0 +1,366 @@
+    private SiddhiAppContext siddhiAppContext;
+    private Map<String, Table> tableMap;
+    private static ScheduledFuture scheduledFuture;
+    private static final String INTERNAL_AGG_TIMESTAMP_FIELD = "AGG_TIMESTAMP";
+
+
+    public void init(AggregationDefinition aggregationDefinition, StreamEventPool streamEventPool,
+                     Map<TimePeriod.Duration, Table> aggregationTables, SiddhiAppContext siddhiAppContext,
+                     Map<String, Table> tableMap) {
+        this.siddhiAppContext = siddhiAppContext;
+        List<Annotation> annotations = aggregationDefinition.getAnnotations();
+        this.tableMap = tableMap;
+        this.streamEventPool = streamEventPool;
+        this.aggregationTables = aggregationTables;
+        for (TimePeriod.Duration duration : aggregationTables.keySet()) {
+            switch (duration) {
+                case SECONDS:
+                    retentionPeriods.put(duration, sec);
+                    break;
+                case MINUTES:
+                    retentionPeriods.put(duration, mins);
+                    break;
+                case HOURS:
+                    retentionPeriods.put(duration, hours);
+                    break;
+                case DAYS:
+                    retentionPeriods.put(duration, days);
+                    break;
+                case MONTHS:
+                    retentionPeriods.put(duration, months);
+                    break;
+                case YEARS:
+                    retentionPeriods.put(duration, years);
+            }
+        }
+
+        Map<String, Annotation> annotationTypes = new HashMap<>();
+
+        for (Annotation annotation : annotations) {
+            annotationTypes.put(annotation.getName().toLowerCase(), annotation);
+        }
+        if (annotationTypes.keySet().contains(SiddhiConstants.NAMESPACE_PURGE)) {
+            if (Objects.nonNull(annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                    getElement(SiddhiConstants.ANNOTATION_ELEMENT_ENABLE))) {
+                String enable = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                        getElement(SiddhiConstants.ANNOTATION_ELEMENT_ENABLE);
+                if (!(enable.equals("true") || enable.equals("false"))) {
+                    throw new SiddhiAppCreationException("Undefined value for enable: " + enable + "." +
+                            " Please use true or false");
+                } else {
+                    purgingEnabled = Boolean.parseBoolean(enable);
+                }
+            }
+            if (purgingEnabled) {
+                // If interval is defined, default value of 15 min will be replaced by user input value
+                if (Objects.nonNull(annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
+                        getElement(SiddhiConstants.NAMESPACE_INTERVAL))) {
+                    String interval = annotationTypes.get(SiddhiConstants.NAMESPACE_PURGE).
```

Comment:
What if the interval is not specified? We should throw an error?

Reply:
As we discussed earlier, if the interval is not defined we will use a default interval to scheduled the purging task. so interval is optional.